pkg/mutation/assignmetadata.go

```diff
@@ -0,0 +1,71 @@
+package mutation
+
+import (
+	"fmt"
+
+	"github.com/open-policy-agent/gatekeeper/pkg/mutation/path/parser"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// MetadataMutator is a mutator wrapping the AssignMetadata type.
+type AssignMetadataMutator struct {
+	//assignMetadata *mutationsv1.AssignMetadata
+	Path  parser.Path
+	Value string
+}
+
+func (m AssignMetadataMutator) Matches(obj metav1.Object, ns *corev1.Namespace) bool {
+	return false
+}
+
+func (m AssignMetadataMutator) Mutate(obj *unstructured.Unstructured) error {
+	return mutate(obj.Object, m.Path.Nodes, m.Value)
+}
+
+func mutate(current interface{}, remainingPaths []parser.Node, value string) error {
+	pathEntry := remainingPaths[0]
+	if len(remainingPaths) == 1 {
+		// TODO: handle types other than object
+		current.(map[string]interface{})[pathEntry.(parser.Object).Reference] = value
+		return nil
+	}
+	switch t := pathEntry.Type(); t {
+	case parser.ObjectNode:
+		next, ok := current.(map[string]interface{})[pathEntry.(parser.Object).Reference]
+		if !ok {
+			// TODO: Add pathTests and add path if required.
+			// next := NewEmptyOfCorrectType(remainingPaths[1])
+			return fmt.Errorf("Path specified by PathEntry does not exist for resource")
+		}
+		if err := mutate(next, remainingPaths[1:], value); err != nil {
```

Comment:
I think this will work without explicit assignment, though we'll definitely need to test this.

Reply:
The test return the incominng obj modified correctly ([here](https://github.com/open-policy-agent/gatekeeper/pull/937/files#diff-7f9a8f1444a9db053bf606ba4a382c424ea5a66f8666d2a2c440a2e6b17f724d)), so there should be no problem.